charms/worker/k8s/src/charm.py

```diff
@@ -125,7 +129,7 @@ def __init__(self, *args):
         self.labeler = LabelMaker(
             self, kubeconfig_path=self._internal_kubeconfig, kubectl=KUBECTL_PATH
         )
-        self._stored.set_default(removing=False)
+        self._stored.set_default(is_dying=False, cluster_name="")
```

Comment:
a unit should keep in stored state the first cluster-name is joins,  It should block if its ever asked to join a different cluster name.